File: benchmarks/java/set/Clojure_Set/SetElementOfSmall2.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;

public class SetElementOfSmall2 {






    private BInteger counter;
    private BSet set;

    public SetElementOfSmall2() {
        counter = (BInteger) new BInteger(0);
        set = (BSet) new BSet(new BInteger(1));
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000)).and(set.elementOf(new BInteger(1)))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        SetElementOfSmall2 exec = new SetElementOfSmall2();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: microbenchmarks/cpp/set/SetElementOfSmall2.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BSet.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef SetElementOfSmall2_H
#define SetElementOfSmall2_H

using namespace std;

class SetElementOfSmall2 {



    private:



        BInteger counter;
        BSet<BInteger > set;

    public:

        SetElementOfSmall2() {
            counter = (BInteger(0));
            set = (BSet<BInteger >((BInteger(1))));
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))._and(set.elementOf((BInteger(1))))).booleanValue()) {
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    SetElementOfSmall2 exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

